### ISSUE TITLE
Incorrect eslint config being applied

### ISSUE THREAD

**Issue by thany** [storybookjs/storybook#12884]
While building or running storybook, we're getting a ton of errors from eslint, which are not relevant to us. This is because storybook is using a different eslint config than the one in my project. Even disabling eslint in our craco config is completely ignored.

Building storybook using the `--debug-webpack` flag, reveals the eslint rule being applied:

```js
{
  test: /\.(js|mjs|jsx|ts|tsx)$/,
  enforce: 'pre',
  use: [
    {
    options: {
      cache: true,
      formatter: 'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\node_modules\\react-scripts\\node_modules\\react-dev-utils\\eslintFormatter.js',
      eslintPath: 'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\node_modules\\eslint\\lib\\api.js',
      resolvePluginsRelativeTo: 'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\node_modules\\react-scripts\\config',
      ignore: false,
      baseConfig: {
      extends: [
        'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\node_modules\\eslint-config-react-app\\index.js'
      ]
      },
      useEslintrc: false
    },
    loader: 'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\node_modules\\eslint-loader\\dist\\cjs.js'
    }
  ],
  include: [
    'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\src',
    'C:\\workspaces\\mijn-enexis\\Sources\\Solution\\.storybook'
  ]
}
```

This is completely incorrect. First of all, its base config is assumed to be craco's base eslint config. This should be overruled by an existing `.eslintrc` file (not extended - **overruled**). In our erlintrc, we are extending from the airbnb config. The craco eslint config we don't use at all. Why is it being forced upon us here?

As a result, not only are we getting a ton of eslint errors, we are also getting them as fatal errors, so building storybook fails.

It gets weirder though. When I rename our `/.storybook` to `/config/storybook` and build storybook with the `-c ./config/storybook` flag, it spews out the same metric ton of errors, but as **warnings**, so building storybook just carries on and succeeds. This makes no sense to me.

Actually tbh, I had my storybook config in `/config/storybook` but I moved it to `/.storybook` hoping it would fix those warnings and make it pick up my `.eslintrc`.

And as icing on the cake, completely disabling eslint in the craco config, is completely ignored. Eslint is executed anyway.

Questions:

1. Why is Storybook using this particular eslint config? What's the reasoning behind setting it up like this?
2. How do I overrule this behaviour?
3. Is this fixed in Storybook 6?

TLDR, what I want to achieve:

1. Have build-storybook pick up our eslint config.
2. Not having it apply *any* other eslint config.
3. Warnings be warnings, not errors.

I'm on Storybook 5.3 still, so I understand this won't be fixed on the Storybook side, which is why I would rather like a workaround of sorts. If this is fixed in Storybook 6, that'll be good to know when we can do the upgrade.

**Comment by MariaNicolosi** [comment]
I solved this by adding the following to the .eslintrc.js file...

  ignorePatterns: [
    'generated-stories-entry.js',
    'storybook-init-framework-entry.js',
  ],

**Comment by valentinpalkovic** [comment]
Thanks for the report. This was a CRA / `react-scripts` preset artifact injecting `eslint-loader` into Storybook 5.3 builds, with `build-storybook` (replaced by `storybook build` in Storybook 7). Modern Storybook no longer injects `eslint-loader`, and CRA itself has been deprecated. Closing as stale.
